src/iam.ts

```diff
@@ -215,11 +215,29 @@ export class IAM extends IAMBase {
     includeClaims = true
   }: { did?: string; includeClaims?: boolean } | undefined = {}) {
     if (this._cacheClient && did) {
-      const didDoc = await this._cacheClient.getDidDocument({ did, includeClaims });
-      return {
-        ...didDoc,
-        service: didDoc.service as (IServiceEndpoint & ClaimData)[]
-      };
+      try {
+        const didDoc = await this._cacheClient.getDidDocument({ did, includeClaims });
+        return {
+          ...didDoc,
+          service: didDoc.service as (IServiceEndpoint & ClaimData)[]
+        };
+      } catch (err) {
+        if (err?.response?.status === 404) {
+          if (this._resolver) {
+            const document = await this._resolver.read(did);
+            return {
+              ...document,
+              service: includeClaims
+                ? ((await this.downloadClaims({
+                    services:
+                      document.service && document.service.length > 0 ? document.service : []
+                  })) as (IServiceEndpoint & ClaimData)[])
+                : []
+            };
+          }
+        }
```

Comment:
Instead of duplicating the code here, I would just fall through to the code below, I think it is the same.

Reply:
👍